core/src/main/scala/kafka/tools/ConsumerPerformance.scala

```diff
@@ -105,16 +105,14 @@ object ConsumerPerformance extends LazyLogging {
     var messagesRead = 0L
     var lastBytesRead = 0L
     var lastMessagesRead = 0L
-    var joinStart = 0L
     var joinTimeMsInSingleRound = 0L
 
     consumer.subscribe(topics.asJava, new ConsumerRebalanceListener {
       def onPartitionsAssigned(partitions: util.Collection[TopicPartition]): Unit = {
-        joinTime.addAndGet(System.currentTimeMillis - joinStart)
```

Comment:
It seems to me that the origin code want to count the elapsed time of joining group. Hence, the ```joinStart``` is required and the following ```joinStart = System.currentTimeMillis``` is required too. 

For another, the initial value of ```joinStart``` is incorrect. It should be equal to ```testStartTime```

Reply:
"In eager rebalancing, it will always be called at the start of a rebalance and after the consumer stops fetching data. In cooperative rebalancing, it will be called at the end of a rebalance on the set of partitions being revoked iff the set is non-empty" is said in https://kafka.apache.org/24/javadoc/org/apache/kafka/clients/consumer/ConsumerRebalanceListener.html#onPartitionsAssigned-java.util.Collection-.

Reply:
I added logs in methods onPartitionsAssigned and onPartitionsRevoked and checked both of them are called just once in normal cases.

Reply:
@chia7712 So I think `joinStart = System.currentTimeMillis` in onPartitionsAssigned is no required, where joinStart is close to endMs.

Reply:
hmmm, is there a good place to update the  start time of join?

Reply:
@chia7712 I don't find a good place to update it. Perhaps we should remove this metric. The way it gets fetchTimeInMs value is also not a good way.   `val fetchTimeInMs = (endMs - startMs) - joinGroupTimeInMs.get` .  There might be some time waiting for connection.

Reply:
@guozhangwang Sorry for my late reply. I am not working on this PR.  Should I abandon this PR? So other people can work on this issue.

Reply:
This has been patched:
https://github.com/apache/kafka/commit/7ceb34bbeccab1d16def1812aecd61a6a3f021e0